File: src/Repository/IRepository.java
package Repository;

import Model.ProgramState;

import java.util.List;

public interface IRepository {
    void add(ProgramState state);
    void logProgramStateExec(ProgramState state);
    List<ProgramState> getProgramList();
    void setProgramList(List<ProgramState> list);
}

File: Toy-Language-C#-Intepreter/Toy-Language-CS-Interpreter/Toy-Language-CS-Interpreter/Repository/IRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language_CS_Interpreter.Model;

namespace Toy_Language_CS_Interpreter.Repository
{
    interface IRepository
    {
        ProgramState GetCurrent();
        void LogProgramState();
    }
}
